StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1214ReadonlyElementsMustAppearBeforeNonReadonlyElements.cs

```diff
@@ -0,0 +1,150 @@
+            }
+
+            var typeDeclaration = (TypeDeclarationSyntax)context.Node;
+
+            FieldDeclarationSyntax previousMember = null;
+            var previousFieldConst = true;
+            var previousFieldStatic = false;
+            var previousFieldReadonly = false;
+            var previousAccessLevel = AccessLevel.NotSpecified;
+            foreach (var member in typeDeclaration.Members)
+            {
+                FieldDeclarationSyntax field = member as FieldDeclarationSyntax;
+                if (field == null)
+                {
+                    previousMember = null;
+                    continue;
+                }
+
+                var modifiers = member.GetModifiers();
+                var currentAccessLevel = MemberOrderHelper.GetAccessLevelForOrdering(member, modifiers);
+                bool currentFieldConst = modifiers.Any(SyntaxKind.ConstKeyword);
+                bool currentFieldStatic = modifiers.Any(SyntaxKind.StaticKeyword);
+                bool currentFieldReadonly = modifiers.Any(SyntaxKind.ReadOnlyKeyword);
+                if (previousMember == null)
+                {
+                    previousMember = field;
+                    previousFieldConst = currentFieldConst;
+                    previousFieldStatic = currentFieldStatic;
+                    previousFieldReadonly = currentFieldReadonly;
+                    previousAccessLevel = currentAccessLevel;
+                    continue;
+                }
+
+                bool compareReadonly = true;
+                for (int j = 0; j < readonlyIndex; j++)
+                {
+                    switch (elementOrder[j])
+                    {
+                    case OrderingTrait.Kind:
+                        // This analyzer only ever looks at sequences of fields.
+                        continue;
+
+                    case OrderingTrait.Accessibility:
+                        if (previousAccessLevel != currentAccessLevel)
+                        {
+                            compareReadonly = false;
+                        }
+
+                        continue;
+
+                    case OrderingTrait.Constant:
+                        if (previousFieldConst || currentFieldConst)
+                        {
+                            compareReadonly = false;
+                        }
+
+                        continue;
+
+                    case OrderingTrait.Static:
+                        if (previousFieldStatic != currentFieldStatic)
+                        {
+                            compareReadonly = false;
+                        }
+
+                        continue;
+
+                    case OrderingTrait.Readonly:
+                    default:
+                        continue;
+                    }
+                }
+
+                if (compareReadonly)
+                {
+                    if ((currentFieldReadonly || currentFieldConst) && !previousFieldReadonly && !previousFieldConst)
```

Comment:
:exclamation: I think that constants should not be considered here. They are handled by SA1203 exclusively.